engine/access/subscription/data_provider.go

```diff
@@ -0,0 +1,71 @@
+package subscription
+
+import (
+	"context"
+	"fmt"
+)
+
+// DataProvider represents a source of sequential items to be streamed to a client.
+//
+// TODO: It can and should operate on generic type  rather than `any`. IT'd require lots of refactoring,
+// so it should be done in a PR with the subscription package refactoring.
+type DataProvider interface {
+	// NextData retrieves the next sequential item from the data source.
+	//
+	// - Retrieves the next available item in sequence and returns it as the first result.
+	// - If no further item is currently available yet (e.g. the next block/data is not ready), it
+	//   should return a non-nil error that satisfies [ErrBlockNotReady]. The streamer interprets this
```

Comment:
Generally, interface docs should document the behavior, not the implementation requirements. Frame it as "Method does X" instead of "Method should be implemented to do X"

```suggestion
	//   [ErrBlockNotReady] is returned. The streamer interprets this
```